tests/validate.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -eo pipefail
+
+COMMANDS=(
```

Comment:
I wonder if these and the `KEYS` could be taken in as argument instead of declared here in the file :thinking: 

Not a necessary change, just an idea :thought_balloon: